pkg/v1/builder/template/plugintemplates/gomod.tmpl

```diff
@@ -1,13 +1,286 @@
 module {{ .RepositoryName }}
 
-go 1.16
+go 1.17
+
+replace (
+	github.com/k14s/kbld => github.com/anujc25/carvel-kbld v0.31.0-update-vendir
+	sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.0.1
+)
 
 require (
 	github.com/aunum/log v0.0.0-20200821225356-38d2e2c8b489
+	github.com/golangci/golangci-lint v1.43.0
+	github.com/spf13/cobra v1.3.0
+	github.com/vmware-tanzu/tanzu-framework v1.4.0-pre-alpha-2.0.20220119234741-712b1ad45604
+)
+
+require (
+	4d63.com/gochecknoglobals v0.1.0 // indirect
```

Comment:
was the old gomod.tmpl way off? Just trying to understand why there is such a huge increase in the dependencies.

Reply:
It's because of Go 1.17, which includes a new `require` block with all the indirect dependencies. I believe that's part of the "module graph" changes in Go.

Reply:
As far as the changes go, there are some major changes in go 1.17 related to module graphs: https://go.dev/ref/mod#graph-pruning

Reply:
ah, thanks both.